### PR TITLE
Replace fscanf with faster parsing #2

### PR DESCRIPTION
**Pull request by KaelaSavia**
Credits to original pull https://github.com/alliedmodders/metamod-source/pull/127 @PeakKS 

Introduces faster parsing to ModuleInMemory.

### CODE REVIEW

**Comment by dvander** [comment]
Do you have any benchmark data for this?

**Comment by KaelaSavia** [comment]
PR replaces fscanf with getline which is faster than fscanf.

Elapsed CPU Time (FSCANF) = '7.91713' sec.
Elapsed CPU Time per Iteration (FSCANF, 500000) = '1.583425e-05' sec.
Elapsed CPU Time (GETLINE) = '4.13407' sec.
Elapsed CPU Time per Iteration (GETLINE, 500000) = '8.268138e-06' sec.

As per in-game performance, on stress test map with a lot of sdkhooked entities. Time goes from 1975 ms to 903 ms. (https://github.com/alliedmodders/sourcemod/issues/1935)

File: core/sourcehook/sh_memory.h [hunk 1]
```diff
@@ -262,37 +262,34 @@ namespace SourceHook
...
-				while (fscanf(pF, "%lx-%lx", &rlower, &rupper) != EOF)
-				{
+				// lower	upper	prot	 stuff				 path
+				// 08048000-0804c000 r-xp 00000000 03:03 1010107	/bin/cat
+				unsigned long rlower, rupper;
+				char *buffer = NULL;
+				size_t bufsize = 0;
+				while (getline(&buffer, &bufsize, pF) != -1) {
+					char *addr_split;
+					rlower = strtoul(buffer, &addr_split, 16);
```
Comment (dvander) [inline core/sourcehook/sh_memory.h:281]: unsigned long rlower = ...
unsigned long rupper = ...

(move variable declarations inside loop)

File: core/sourcehook/sh_memory.h [hunk 2]
```diff
@@ -262,37 +262,34 @@ namespace SourceHook
 		{
 #if SH_SYS == SH_SYS_LINUX
 			// On linux, first check /proc/self/maps
-			long lower = reinterpret_cast<long>(addr);
-			long upper = lower + len;
-
+			unsigned long lower = reinterpret_cast<unsigned long>(addr);
+			unsigned long upper = lower + len;
+			
+			bool bFound = false;
```
Comment (dvander) [inline core/sourcehook/sh_memory.h:268]: Move "bFound" declaration inside the "if (pF)"

**Review by dvander — COMMENTED** [review]
Good enough for me! Modulo a few nits.

**Review by psychonic — APPROVED** [review]

**Pull request merged**